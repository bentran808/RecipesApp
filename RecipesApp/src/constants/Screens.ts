export default {
  Main: { name: 'Main' },
  Home: { name: 'Home', label: 'HOME' },
  Categories: { name: 'Categories', label: 'CATEGORIES' },
  Recipe: { name: 'Recipe' },
  RecipesList: { name: 'RecipesList' },
  Ingredient: { name: 'Ingredient' },
  IngredientsDetails: { name: 'IngredientsDetails' },
  Search: { name: 'Search', label: 'SEARCH' }
};
