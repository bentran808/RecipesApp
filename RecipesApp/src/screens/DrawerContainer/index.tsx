import { DrawerNavigationHelpers } from '@react-navigation/drawer/lib/typescript/src/types';
import MenuItem from 'components/MenuItem';
import Screens from 'constants/Screens';
import React, { useCallback } from 'react';
import { View } from 'react-native';
import { CategoryIcon, HomeIcon, SearchIcon } from 'theme';
import styles from './styles';

type Props = {
  navigation: DrawerNavigationHelpers;
};

const DrawerContainer = ({ navigation }: Props) => {
  const handlePressHome = useCallback(() => {
    navigation.navigate(Screens.Home.name);
    navigation.closeDrawer();
  }, []);

  const handlePressCategories = useCallback(() => {
    navigation.navigate(Screens.Categories.name);
    navigation.closeDrawer();
  }, []);

  const handlePressSearch = useCallback(() => {
    navigation.navigate(Screens.Search.name);
    navigation.closeDrawer();
  }, []);

  const drawerList = [
    {
      title: Screens.Home.label,
      source: HomeIcon,
      onPress: handlePressHome
    },
    {
      title: Screens.Categories.label,
      source: CategoryIcon,
      onPress: handlePressCategories
    },
    {
      title: Screens.Search.label,
      source: SearchIcon,
      onPress: handlePressSearch
    }
  ];

  return (
    <View style={styles.content}>
      <View style={styles.container}>
        {drawerList.map(({ title, source, onPress }, index) => (
          <MenuItem key={index} title={title} source={source} onPress={onPress} />
        ))}
      </View>
    </View>
  );
};

export default DrawerContainer;
