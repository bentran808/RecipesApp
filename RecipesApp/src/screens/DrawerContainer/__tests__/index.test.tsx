import Screens from 'constants/Screens';
import { TouchableHighlight } from 'react-native';
import renderer from 'react-test-renderer';
import DrawerContainer from 'screens/DrawerContainer';

describe('Drawer Container', () => {
  let navigation: any;

  beforeEach(() => {
    navigation = {
      navigate: jest.fn(),
      closeDrawer: jest.fn()
    };
  });

  test('should render correctly', () => {
    const tree = renderer.create(<DrawerContainer navigation={navigation} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should call function handlePressHome', () => {
    const component = renderer.create(<DrawerContainer navigation={navigation} />);
    const button = component.root.findAllByType(TouchableHighlight)[0];
    button.props.onPress();
    expect(navigation.navigate).toHaveBeenCalledWith(Screens.Home.name);
  });

  test('should call function handlePressCategories', () => {
    const component = renderer.create(<DrawerContainer navigation={navigation} />);
    const button = component.root.findAllByType(TouchableHighlight)[1];
    button.props.onPress();
    expect(navigation.navigate).toHaveBeenCalledWith(Screens.Categories.name);
  });

  test('should call function handlePressSearch', () => {
    const component = renderer.create(<DrawerContainer navigation={navigation} />);
    const button = component.root.findAllByType(TouchableHighlight)[2];
    button.props.onPress();
    expect(navigation.navigate).toHaveBeenCalledWith(Screens.Search.name);
  });
});
