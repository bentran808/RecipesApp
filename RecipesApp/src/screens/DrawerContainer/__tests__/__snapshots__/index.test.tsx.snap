// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer Container should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "marginVertical": 5,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 25,
              "width": 25,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 16,
              "marginLeft": 10,
              "marginTop": 2,
            }
          }
        >
          HOME
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "marginVertical": 5,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 25,
              "width": 25,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 16,
              "marginLeft": 10,
              "marginTop": 2,
            }
          }
        >
          CATEGORIES
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "marginVertical": 5,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 25,
              "width": 25,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 16,
              "marginLeft": 10,
              "marginTop": 2,
            }
          }
        >
          SEARCH
        </Text>
      </View>
    </View>
  </View>
</View>
`;
